File: ss-won/programmers/64065.js
function solution(s) {
  let answer = [];
  s = s.substring(1, s.length - 1);
  let open = s.indexOf("{");
  let close = s.indexOf("}");
  let block;
  let arr = [];

  let ck = function (v) {
    v = parseInt(v);
    for (let val of answer) {
      if (parseInt(val) == v) return false;
    }
    return true;
  };

  //{,}기준으로 나눠서 배열에 넣기
  while (open != -1 || close != -1) {
    block = s.substring(open + 1, close);
    arr.push(block.split(","));
    open = s.indexOf("{", close + 1);
    close = s.indexOf("}", close + 1);
  }
  //length크기만큼 오름차순 정렬
  arr.sort((a, b) => a.length - b.length);

  arr.forEach((v) => {
    v.forEach((val) => {
      if (ck(val)) answer.push(parseInt(val));
    });
  });

  return answer;
}

File: ss-won/programmers/64065.py
def getTuples(s):
    tuples, s, i = [], s[1:len(s)-1], 0
    while i < len(s):
        if s[i] == '{':
            tp, i, j = [], i+1, i+1
            while (1):
                if s[i] == ',' or s[i] == '}':
                    tp.append(int(s[j:i]))
                    if s[i] == '}':
                        break
                    j = i+1
                i += 1
            tuples.append(tp)
        i += 1
    return tuples


def solution(s):
    answer = []
    tuples = sorted(getTuples(s), key=lambda x: len(x))
    for tp in tuples:
        for i in range(len(tp)):
            if not tp[i] in answer:
                answer.append(tp[i])
    return answer
